### PR TITLE
CASMINST-6608 Build for NOOS

### PR DESCRIPTION
**Pull request by rustydb**
At the moment, iuf-cli does not build for SP5. Since this is a tedious aspect to maintain, I've gone ahead and just switch iuf-cli to publish to noos.

This does factor in some Jenkinsfile, Makefile, and Spec changes. I used a template I was already using to speed up the implementation.

I created a test RPM here: https://artifactory.algol60.net/artifactory/csm-rpms/hpe/unstable/noos/iuf-cli/x86_64/iuf-cli-1.6.1.dev1%2Bg52bef92-1.x86_64.rpm

That test RPM I installed and was able to invoke its callable:

```bash
redbull-pit:~ # iuf --help
usage: iuf [-h] [-i INPUT_FILE] [-w] [-a ACTIVITY] [-c CONCURRENCY] [-b BASE_DIR] [-s STATE_DIR] [-m MEDIA_DIR] [--log-dir LOG_DIR]
           [-l {CRITICAL,ERROR,WARNING,INFO,DEBUG,TRACE}]
           {run,activity,list-stages|ls,resume,restart,abort,list-activities|la,workflow} ...

The CSM Install and Upgrade Framework (IUF) CLI.

options:
  -h, --help            show this help message and exit
  -i INPUT_FILE, --input-file INPUT_FILE
                        YAML input file used to provide arguments to `iuf`. Command line arguments will override entries in the input
                        file. Can also be set via the IUF_INPUT_FILE environment variable.
  -w, --write-input-file
                        Create an input file for iuf populated with the command line options specified and exit. This input file can be
                        specified with the `-i` option on subsequent runs. Using an input file simplifies iuf commands with many options.
                        Note that the general iuf command does not change; so for a long iuf command, add this flag to the command to
                        write the input file.
  -a ACTIVITY, --activity ACTIVITY
                        Activity name. Must be a unique identifier. Activity names must contain only lowercase letters (a-z), numbers
                        (0-9), periods (.), and dashes (-). Can also be set via the IUF_ACTIVITY environment variable.
  -c CONCURRENCY, --concurrency CONCURRENCY
                        During stage processing Argo runs workflow steps in parallel. By default up to 10 steps will be executed
                        simultaneously. Use `--concurrency N` to decrease the limit to N. Increasing this limit is not recommended.
  -b BASE_DIR, --base-dir BASE_DIR
                        Base directory for state and log file directories. Defaults to ${RBD_BASE_DIR}/iuf/[activity], where
                        ${RBD_BASE_DIR} is /etc/cray/upgrade/csm.
  -s STATE_DIR, --state-dir STATE_DIR
                        A directory used to store the current state of stages, used by `iuf` but primarily not of interest to users.
                        Defaults to [base-dir]/state.
  -m MEDIA_DIR, --media-dir MEDIA_DIR
                        Location of installation media to be used. Defaults to ${RBD_BASE_DIR}/[activity], where ${RBD_BASE_DIR} is
                        /etc/cray/upgrade/csm. `iuf` cannot access installation media outside of ${RBD_BASE_DIR}, however input files
                        provided by other `iuf` arguments can exist outside of ${RBD_BASE_DIR}.
  --log-dir LOG_DIR     Location used to store log files. Defaults to [base-dir]/log.
  -l {CRITICAL,ERROR,WARNING,INFO,DEBUG,TRACE}, --level {CRITICAL,ERROR,WARNING,INFO,DEBUG,TRACE}
                        Set the log message level that determines what is displayed on `iuf` standard output. Messages of this level or
                        higher are displayed.

subcommands:
  {run,activity,list-stages|ls,resume,restart,abort,list-activities|la,workflow}
```

### CODE REVIEW

**Review by JohnNavitskyHPE — APPROVED** [review]

File: Jenkinsfile.github [hunk 1]
```diff
@@ -25,117 +25,139 @@
...
+    return
+} else if (env.BRANCH_NAME ==~ ~"^dependabot/github_actions") {
+    currentBuild.result = 'SUCCESS'
+    echo "Dependabot GitHub action builds are skipped."
+    return
 }
+
+// Only consider X.Y.Z and X.Y.Z.postN tags as stable.
+// Never consider X.Y.Z{[a|b|rc} or X.Y.Z.* tags (that are not post-releases) as stable.
+// The ==~ operator performs an exact match.
```
Comment (kburns-hpe) [inline Jenkinsfile.github:43]: Is this comment needed? If so, I don't see ==~ being used below.

**Review by kburns-hpe — APPROVED** [review]

File: Jenkinsfile.github (hunk 1 above)
Comment (rustydb) [reply Jenkinsfile.github:43]: I copied it from another build, let me fix.

Comment (rustydb) [reply Jenkinsfile.github:43]: Used two lines below this one.

Comment (rustydb) [reply Jenkinsfile.github:43]: ` 45 def isStable = (env.TAG_NAME != null & env.TAG_NAME ==~ stableToken) ? true : false`

File: Makefile [hunk 2]
```diff
@@ -22,43 +22,164 @@
...
 
 rpm_build:
-	NAME=${NAME} VERSION=${VERSION} rpmbuild -ba $(SPEC_FILE) --define "_topdir $(BUILD_DIR)"
+	rpmbuild -ba $(BUILD_DIR)/SPECS/$(SPEC_FILE) --target ${ARCH} --define "_topdir $(BUILD_DIR)"
 
-venv:
-	${PYTHON} -m venv ${VENV}
-	. ${VENV}/bin/activate; pip install -r requirements.txt
+#############################################################################
+# RPM targets
```
Comment (haasken-hpe) [inline Makefile:173]: These are image targets.

File: Makefile [hunk 3]
```diff
@@ -22,43 +22,164 @@
...
+image_login:
+	docker login artifactory.algol60.net
+
+image_build:
+	docker build -t artifactory.algol60.net/csm-docker/stable/craycli/swagger2openapi:latest utils/
+
+image_publish: image_build
+	docker push artifactory.algol60.net/csm-docker/stable/craycli/swagger2openapi:latest
+
+image: image_publish
```
Comment (haasken-hpe) [inline Makefile:185]: These look like they were copied verbatim from `craycli` and they do not apply here.

File: Makefile [hunk 4]
```diff
@@ -22,43 +22,164 @@
...
+	@echo
+	@echo '    prepare              Prepare for making an RPM.'
+	@echo '    rpm_build            Builds the RPM.'
+	@echo '    rpm_build_source		Builds the SRPM.'
+	@echo '    rpm_package_source   Creates the RPM source tarball.'
+	@echo
+	@echo '    image_login   		Logs into the Docker registry for pulling and publishing images.'
+	@echo '    image_build   		Builds the Swagger testing image.'
+	@echo '    image_publish   		Builds and publishes the testing image.'
+	@echo ''
```
Comment (haasken-hpe) [inline Makefile:138]: remove these too

**Review by haasken-hpe — COMMENTED** [review]
@rustydb, please review the `Makefile` and `Jenkinsfile.github` changes in my PR https://github.com/Cray-HPE/iuf-cli/pull/160/files

I had to rebase, and I want to make sure I did it right. I also removed what I think are erroneous inclusions in the `Makefile` here.

**Pull request merged**